src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PlacementRequestsController.kt

```diff
@@ -192,7 +194,11 @@ class PlacementRequestsController(
     }
 
     val result = extractEntityFromAuthorisableActionResult(
-      placementRequestService.withdrawPlacementRequest(id, user, reason),
+      placementRequestService.withdrawPlacementRequest(
+        id,
+        user,
+        reason,
+        checkUserPermissions = true),
```

Comment:
I wonder if it would be clearer about when and why it's necessary to check the user's permission if this optional parameter were named something like `requestedByUser` or `cascadedWithdrawal`?

Reply:
This was actually on my TODO list, great minds think alike! :) In a subsequent PR i've added a WithdrawalContext that is passed to all withdrawal functions:

`data class WithdrawalContext(
  val triggeringUser: UserEntity?,
  val triggeringEntityType: WithdrawableEntityType,
)

enum class WithdrawableEntityType {
  Application,
  PlacementRequest,
  PlacementApplication,
  Booking,
}
`

The 'triggeringEntityType' can be used to determine if this was a 'direct' request from a user or a cascade. This means we can then remove the 'checkUserPermissions' boolean and use that instead. I'll make that change now and queue it up for review once this is merged 👍